arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -1537,8 +1680,9 @@ Result fromArrayComparison(irs::boolean_filter*& filter,
     case SCOPED_VALUE_TYPE_ARRAY: {
       size_t const n = value.size();
       Result buildRes;
+      size_t matchCount;
       aql::AstNodeType arrayExpansionNodeType;
-      std::tie(buildRes, arrayExpansionNodeType) =
+      std::tie(buildRes, arrayExpansionNodeType, matchCount) =
```

Comment:
nit: could be simplified by using `auto [buildRes, arrayExpansionNodeType, matchCount] = ...`
```suggestion
      std::tie(buildRes, arrayExpansionNodeType, matchCount) =
```